tests/unit/s2n_server_key_share_extension_test.c

```diff
@@ -0,0 +1,356 @@
+                }
+            }
+
+            EXPECT_NOT_NULL(server_conn->secure.server_ecc_params.negotiated_curve);
+            EXPECT_EQUAL(server_conn->secure.server_ecc_params.negotiated_curve->iana_id, s2n_ecc_supported_curves[i].iana_id);
+
+            /* Server sends ServerHello key_share */
+            EXPECT_SUCCESS(s2n_extensions_server_key_share_send(server_conn, &server_hello_key_share));
+
+            /* Client receives ServerHello key_share */
+            S2N_STUFFER_READ_EXPECT_EQUAL(&server_hello_key_share, TLS_EXTENSION_KEY_SHARE, uint16);
+            S2N_STUFFER_READ_EXPECT_EQUAL(&server_hello_key_share, s2n_extensions_server_key_share_send_size(server_conn) - 2, uint16);
+            EXPECT_SUCCESS(s2n_extensions_server_key_share_recv(client_conn, &server_hello_key_share));
+            EXPECT_EQUAL(s2n_stuffer_data_available(&server_hello_key_share), 0);
+
+            EXPECT_EQUAL(server_conn->secure.server_ecc_params.negotiated_curve, client_conn->secure.server_ecc_params.negotiated_curve);
+
+            /* Ensure both client and server public key matches */
+            s2n_public_ecc_keys_are_equal(&server_conn->secure.server_ecc_params, &client_conn->secure.server_ecc_params);
+            s2n_public_ecc_keys_are_equal(&server_conn->secure.client_ecc_params[i], &client_conn->secure.client_ecc_params[i]);
+
+            /* Server generates shared key based on Server's Key (EC Key) and Client's public key (EC Point) */
+            struct s2n_blob server_shared_secret = { 0 };
+            EXPECT_SUCCESS(s2n_ecc_compute_shared_secret_from_params(
+                &server_conn->secure.server_ecc_params,
+                &server_conn->secure.client_ecc_params[i],
+                &server_shared_secret));
+
+            /* Clients generates shared key based on Client's EC Key and Server's public key */
+            struct s2n_blob client_shared_secret = { 0 };
+            EXPECT_SUCCESS(s2n_ecc_compute_shared_secret_from_params(
+                &client_conn->secure.client_ecc_params[i],
+                &client_conn->secure.server_ecc_params,
+                &client_shared_secret));
+
+            /* Test that server shared secret matches client shared secret */
+            EXPECT_EQUAL(server_shared_secret.size, shared_secret_size[i]);
+
+            S2N_BLOB_EXPECT_EQUAL(server_shared_secret, client_shared_secret);
+
+            EXPECT_SUCCESS(s2n_free(&client_shared_secret));
+            EXPECT_SUCCESS(s2n_free(&server_shared_secret));
+
+            /* Clean up */
+            EXPECT_SUCCESS(s2n_stuffer_free(&client_hello_key_share));
+            EXPECT_SUCCESS(s2n_stuffer_free(&server_hello_key_share));
+            EXPECT_SUCCESS(s2n_connection_free(client_conn));
+            EXPECT_SUCCESS(s2n_connection_free(server_conn));
+        }
+    }
+
+    END_TEST();
+    return 0;
+}
+
+static int s2n_public_ecc_keys_are_equal(struct s2n_ecc_params *params_1, struct s2n_ecc_params *params_2)
```

Comment:
I think we've written this function a couple times for tests now. It's probably worth moving to testlib.